beets/importer.py

```diff
@@ -754,6 +754,7 @@ def add(self, lib):
             self.record_replaced(lib)
             self.remove_replaced(lib)
             self.album = lib.add_album(self.imported_items())
+            self.album.set_parse('data_source', self.imported_items()[0].data_source)
```

Comment:
How about `self.album.data_source = ...` instead of using the `set_parse` method?